docs/publishing/index.md

```diff
@@ -0,0 +1,123 @@
+
+## Do export side effects separately
+
+A [side effect](<https://en.wikipedia.org/wiki/Side_effect_(computer_science)>) in javascript is created when script effects state outside of the scope in which it is run. This includes when code in an es module effects values available on the `window` or `document` as is inherent to the reality of Custom Elements in that `window.customElements.define()` registers an element in the global registry attached to `window`, whether you explicitly include `window` in your code when doing so or not. JS files with side effects should exist separately from your packages entry point. `npm init @open-wc` exemplifies this through its creation of `my-element.js` which manages the registration of the custom element that this command generates.
+
+## Do keep side effects out of your index.js
+
+When not able to keep code with side effects out of your entry point, be sure to contain such commands within a method (e.g. `run()`, `init()`, etc.) to prevent those side effects from occurring at run time.
+
+## Do import 3rd party node modules with "bare" import specifiers
+
+When referring to 3rd party dependencies in your code, you cannot guarantee that implementing projects rely on the same file structure your package as you do. To prevent dependency pointing from causing issue across different environments be sure to import these dependencies into your project with "bare" import specifiers so that tooling in your users' environment can manage resolving the location of those dependencies; e.g. `import { LitElement } from 'lit-element';`
+
+## Do include file extensions in import specifiers, always
+
+When not pointing directly the entry point files referenced by "bar" import specifiers, be sure to reference files (both in your package and your dependencies) with their complete file extension; e.g. `import { ifDefined } from 'lit-html/directives/if-defined.js';`
+
+## Optional: use `type: "module"`
+
+If scripting in your packages is resilient to the realities made true in Node@12+ when using `type: "module"` in your `package.json`, you should use it. If all of the JS in your package is es modules, then this is a cheap win for your package and as it can support tooling the references your package in leveraging it more efficiently, however excluding it is a simple way in the near term to not require this of all of the JS your write.
```

Comment:
we should expand on what 'the realities' are, or link to more information. People reading this may read this, just put type: module, and have their stuff blow up

Reply:
> cheap win for your package and as it can support tooling the references your package in leveraging it more efficiently

this sentence is unclear to me, as well as:

> however excluding it is a simple way in the near term to not require this of all of the JS your write.

Reply:
I was essentially looking to say "if it blows up, don't use it"... @LarsDenBakker you know much more about this than me, any references I could use here, or realities I should address directly?